### PR TITLE
27 room unavailabilities

### LINKED ISSUE #27: Unverfügbarkeit von Räumen
Als Planer möchte ich,
bestimmte Räume für bestimmte Tage oder Zeitfenster sperren können, sodass kein Termin während dieser Zeit in diesen Raum gelegt wird,
weil mir manche Räume (z.B. im Zweitzugriffsrecht) nur zeitfensterweise zugewiesen werden.

### PR DESCRIPTION
**Pull request by roschaefer**
resolves #27 
@felixKubicek review plz.
Atm, it's not possible to see the unavailabilities. One unavailability is created in the seeds.rb

### CODE REVIEW

File: UPtables/script/timetabling.lp [hunk 1]
```diff
@@ -52,6 +52,10 @@ N { assigned(C,D,P) : d(D), ppd(P) } N :- course(C,_,N,_,_,_).
 
 :- not { assigned(C,D,P) : c(C) } N, d(D), ppd(P), rooms(N).
 
+%%% Room Unavailability: If a room is unavailable during a certain time, no
+%%% lecture be given in this room at that time.
+:- assigned(C,R,D,P), room_unavailable(R,D,P).
+
```
Comment (roschaefer) [inline UPtables/script/timetabling.lp]: please note: I was wrong, Mutsunori doesn't have the room unavailability in the original encoding. ```room_constraint``` means a room is unsuitable for a given course. Not for a given time. I had create the constraint myself.

Comment (felixKubicek) [inline UPtables/script/timetabling.lp]: Jes, in the original encoding there are no unavailability constraint for rooms. We will have to decide in the future how to  distinguish between different unavailabilities (room, teacher, ...). We may use something like 
"unavailability(room("H-01"), ..., ....)" or just different predicates for each type of unavailability.

**Comment by felixKubicek** [comment]
Just like with the ASP encoding we have to think about pulling different unavailabilities together.
Teacher unavailabilities are mapped to the Unavailability class (maybe just one Unavailability class with day and timeslot attribute and additionally a polymorphic association to rooms, teachers or whatever )

**Comment by roschaefer** [comment]
@felixKubicek as we talked on the phone: I would follow the [KISS principle](http://en.wikipedia.org/wiki/KISS_principle) and simply create two different models

**Comment by roschaefer** [comment]
@felixKubicek merge plz

**Pull request merged**